Packages/ChatSecure/ChatSecure/Classes/Networking/Services/Messsage/MessageService.swift

```diff
@@ -0,0 +1,449 @@
+//
+//  MessageService.swift
+//  ChatSecure
+//
+//  Created by NamNH on 13/04/2022.
+//
+
+import Foundation
+import Networking
+import Common
+import SignalServiceKit
+import LibSignalClient
+import SwiftSRP
+
+public protocol IMessageService {
+	func sendMessageInPeer(senderId: String,
+						   ownerDomain: String,
+						   receiverId: String,
+						   receiverDomain: String,
+						   groupId: Int64,
+						   plainMessage: String,
+						   isForceProcessKey: Bool,
+						   cachedMessageId: Int) async -> Result<RealmMessage, Error>
+	
+	func sendMessageInGroup(senderId: String,
+							ownerWorkspace: String,
+							groupId: Int64,
+							isJoined: Bool,
+							plainMessage: String,
+							cachedMessageId: Int,
+							isForward: Bool) async -> Result<RealmMessage, Error>
+	
+	func getMessage(ownerDomain: String,
+					ownerId: String,
+					groupId: Int64,
+					loadSize: Int,
+					isGroup: Bool,
+					lastMessageAt: Int64) async -> Result<[RealmMessage], Error>
+	
+	func decryptPeerMessage(senderName: String, message: Data, messageId: String) -> String?
+	func decryptGroupMessage(senderId: String, senderDomain: String, ownerId: String, ownerDomain: String, groupID: Int64, message: Data) async -> String?
+	
+}
+
+public class MessageService {
+	private let clientStore: ClientStore
+	private let signalStore: ISignalProtocolInMemoryStore
+	private let senderStore: ISenderKeyStore
+		
+	public init(clientStore: ClientStore, signalStore: ISignalProtocolInMemoryStore, senderStore: ISenderKeyStore) {
+		self.clientStore = clientStore
+		self.signalStore = signalStore
+		self.senderStore = senderStore
+	}
+}
+
+extension MessageService: IMessageService {
+	public func sendMessageInPeer(senderId: String, ownerDomain: String, receiverId: String, receiverDomain: String, groupId: Int64, plainMessage: String, isForceProcessKey: Bool, cachedMessageId: Int) async -> Result<RealmMessage, Error> {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/Packages/ChatSecure/ChatSecure/Classes/Networking/Services/Messsage/MessageService.swift#L58"></span>Function body should span 50 lines or less excluding comments and whitespace: currently spans 51 lines<br />
<code>function_body_length</code> <code>MessageService.swift:58</code></td>
    </tr>
  </tbody>
</table>